charts/kured/README.md

```diff
@@ -21,8 +21,40 @@ $ helm delete my-release
 
 The command removes all the Kubernetes components associated with the chart and deletes the release.
 
+## Upgrade Notes
+
+### From 2.x to 3.x
+
+The Helm chart labels have been realigned to conform with the [standard labels](https://helm.sh/docs/chart_best_practices/labels/#standard-labels) in the current Helm Chart Best Practices guide, so this upgrade will fail unless the DaemonSet is deleted and recreated. The only way that Helm supports delete and recreate is by uninstalling and reinstalling.
+
+With a GitOps tool (e.g. Flux), the existing HelmRelease has to be deleted first and pruned, then apply the HelmRelease with a second commit and the new version.
+
+\*Unfortunately, [Helm does not support delete and recreate](https://github.com/fluxcd/flux2/discussions/2854#discussioncomment-3057748) when using upgrades, making upgrades requiring an immutable field change impossible.
+
+Supposing users want to enable metrics and use a `ServiceMonitor` with the `kube-prometheus-stack` chart's default `prometheus` instance. Starting with a chart that has values:
+
+```
+metrics:
+  create: true
+  labels:
+    release: kube-prometheus-stack
+updateStrategy: RollingUpdate
+```
+
+The `kube-prometheus-stack` chart, by default, expects any `ServiceMonitor` to have a label `release` which matches the `kube-prometheus-stack` installation's `Release.Name`. Kured was already setting the same `release` label on every YAML resource to its own Helm chart `Release.Name` until this Helm chart v3.x, which made the default configuration for Kured metrics incompatible with the default for `ServiceMonitor`.
+
+Labels on a DaemonSet are immutable so `helm upgrade --force` will also fail:
+
+```
+Error: cannot patch "kured" with kind DaemonSet: DaemonSet.apps "kured" is invalid: spec.selector: Invalid value: v1.LabelSelector{MatchLabels:map[string]string{"release":"kube-prometheus-stack"}, MatchExpressions:[]v1.LabelSelectorRequirement(nil)}: field is immutable
+```
+
+Users do not need to make any configuration change, only to uninstall and reinstall the `kured` chart. Manually deleting the DaemonSet and recreating it via the next upgrade will also work in environments where direct cluster access is permitted.
```

Comment:
(Z 46-52): This can be removed, as this is the Kubernetes behaviour which can't be changed. Maybe one sentence can be added to the previous text-block which shortly descibes, that changing labels on existing objects needs a recreation.

Reply:
It's not that changing labels on existing objects needs a recreation though, it's DaemonSet in particular which is immutable. (Anyone who uses Kured must use it in DaemonSet mode though, so the fact is probably not really necessary to mention.)

Reply:
Label-Selectors are immutable (e.g. in the PodSpec of the DaemonSet)

Reply:
e42317eca41bf8f273be3faa46f349218f399ac7

Reply:
If there is a finer point in the language that I've gotten subtly wrong then it's beyond my understanding, welcome to suggest any corrections or further deletions if you prefer. I'll accept them.